### PR TITLE
Clarify Body package data algorithm with multipart/form-data MIME type

### LINKED ISSUE #392: Clarification of Body package data algorithm with bytes, FormData and multipart/form-data MIME type
The Body package data algorithm with bytes, FormData and multipart/form-data MIME type step 1 says

> Parse bytes, using the value of the `boundary` parameter from MIME type and utf-8 as encoding, per the rules set forth in Returning Values from Forms: multipart/form-data. [RFC2388]

That is not very detailed parsing specification.
While implementing the algorithm for Chromium, I ran into interpretation issues.
My main question is how to parse different kind of multipart parts.

Parts whose Content-Disposition header field contains a filename parameter must surely be parsed into **Blob**s. That is what Mozilla does.

What about multipart parts which have a Content-Type header field but whose Content-Disposition header field does not contain a filename parameter? To my understanding, it is not possible to construct such parts with **FormData** because **USVString**s will never generate a Content-Type header field and **Blob**s have a default file name ("blob"). Should these parts be parsed into **Blob**s in spite of the missing filename parameter?

Should a multipart part with a name "_charset_" have no effect on decoding encoding?
I think that probably yes, because RFC2388 does not even describe that name (but RFC7578 does) and because  the body package data algorithm with bytes, FormData and application/x-www-form-urlencoded MIME type runs the application/x-www-form-urlencoded parser steps with a use_charset_flag implicitly unset thus it does not change decoding encoding based on the "_charset_" component.

If string parts can never have a Content-Type header field, they can never have a charset parameter either. The multipart part with a name "_charset_" does not seem to have an effect on decoding, either. Does it thus follow that string parts must always be decoded using utf-8 encoding?

### PR DESCRIPTION
**Pull request by eehakkin**
Fixes #392.

### CODE REVIEW

**Comment by annevk** [comment]
@youennf what do you think? This seems like an improvement. I have a couple nits I'll push in a commit once everyone is on board with this.

**Comment by yutakahirano** [comment]
lg

**Comment by annevk** [comment]
@youenff ping.

**Comment by annevk** [comment]
@youennf ping.

**Comment by youennf** [comment]
Sorry for the delay, looks like an improvement to me as well!

I would split some sentences in two so that they are more readable.
entry might need a link reference.
And I guess entry has a "value" and not "values"

**Comment by eehakkin** [comment]
I split some sentences and corrected plurality errors.
Bikeshed correctly links <a>entry</a> to https://xhr.spec.whatwg.org/#concept-formdata-entry.
Don't know what happens with Travis.

**Comment by annevk** [comment]
I think you can satisfy Travis by rebasing at this point. Sorry about that.

File: fetch.bs [hunk 1]
```diff
@@ -4363,10 +4363,30 @@ runs the associated steps:
...
+    the `<code>Content-Type</code>` header of the part if the part has such header or
+    `<code>text/plain</code>` (the default defined by [[!RFC7578]] section 4.4) otherwise.</p>
+
+    <p>Each part whose `<code>Content-Disposition</code>` header does not contain
+    a `<code>filename</code>` parameter must be parsed into an <a>entry</a>
+    whose value is a {{USVString}} whose content is the <a>utf-8</a> decoded content of the part.
+    This is done regardless of the presence or the value of a `<code>Content-Type</code>` header
+    and regardless of the presence or the value of a `<code>charset</code>` parameter.</p>
+
+    <p>A part whose `<code>Content-Disposition</code>` header contains a `<code>name</code>`
```
Comment (annevk) [inline fetch.bs]: I think this would be clearer as a note (class=note) since it's just a statement of fact.

File: fetch.bs [hunk 2]
```diff
@@ -4363,10 +4363,30 @@ runs the associated steps:
...
+    a `<code>filename</code>` parameter must be parsed into an <a>entry</a>
+    whose value is a {{File}} whose content is the content of the part. The {{File/name}} attribute
+    of the {{File}} must have the value of the `<code>filename</code>` parameter of the part.
+    The {{Blob/type}} attribute of the {{File}} must have the value of
+    the `<code>Content-Type</code>` header of the part if the part has such header or
+    `<code>text/plain</code>` (the default defined by [[!RFC7578]] section 4.4) otherwise.</p>
+
+    <p>Each part whose `<code>Content-Disposition</code>` header does not contain
+    a `<code>filename</code>` parameter must be parsed into an <a>entry</a>
+    whose value is a {{USVString}} whose content is the <a>utf-8</a> decoded content of the part.
```
Comment (annevk) [inline fetch.bs]: Use "whose value is the <a>UTF-8 decoded</a> content of the part" instead.

Comment (eehakkin) [inline fetch.bs]: Done.

File: fetch.bs (hunk 1 above)
Comment (eehakkin) [inline fetch.bs]: Done.

**Comment by annevk** [comment]
@eehakkin are you okay with the changes I made? Looks pretty good now.

**Comment by eehakkin** [comment]
@annevk: Yes, sure. I am okay with the changes you made.

**Comment by annevk** [comment]
Great, thanks!

**Pull request merged**